hw/bitstream/BUILD

```diff
@@ -53,13 +67,21 @@ config_setting(
     },
 )
 
+config_setting(
+    name = "bitstream_local_splice",
+    define_values = {
+        "bitstream": "local",
+    },
+)
+
 filegroup(
     name = "test_rom",
     testonly = True,
     srcs = select({
         "bitstream_skip": ["skip.bit"],
         "bitstream_vivado": ["//hw/bitstream/vivado:fpga_cw310_test_rom"],
         "bitstream_gcp_splice": [":gcp_spliced_test_rom"],
+        "bitstream_local_splice": [":local_spliced_test_rom"],
```

Comment:
I've been meaning to create a tool that creates a bitstream cache entry from a local build and makes this easier. Perhaps I ought to get on that, haha.

Reply:
From a users perspective, would I then do
```sh
export BAZEL_BITSTREAMS_CACHE=${HOME}/local-bitstreams-cache
export BITSTREAM="--offline latest"
bazel build //hw/bitstream:rom # let Vivado build a bitstream with an ILA
bazel build //hw/bitstream:rom_otp_test_unlocked0 # splice ROM and OTP into bitstream with ILA
./bazelisk.sh outquery-all //hw/bitstream:rom_otp_test_unlocked0 # get path to .bit file I can load with Vivado
```
?

For implementation considerations, @timothytrippel may want to chime in -- he authored the implementation in this PR :-)

Reply:
Yes, that's right. :)

Prior to doing this, you would need your three files from the bitstream build under the bitstreams cache directory:
```
${HOME}/local-bitstreams-cache/${BITSTREAM_KEY}/chip_earlgrey_cw310.bit
${HOME}/local-bitstreams-cache/${BITSTREAM_KEY}/rom.mmi
${HOME}/local-bitstreams-cache/${BITSTREAM_KEY}/otp.mmi
${HOME}/local-bitstreams-cache/latest.txt
```
Where `BITSTREAM_KEY` would be the originating git hash (ideally). If you want to use the special key "latest", then `latest.txt` should contain the text of `${BITSTREAM_KEY}`, like this:

```
echo ${BITSTREAM_KEY} > ${HOME}/local-bitstreams-cache/latest.txt
```

But you don't actually have to update `latest.txt`. You may use `${BITSTREAM_KEY}` instead of "latest" for selecting from the cache:
```diff
-export BITSTREAM="--offline latest"
+export BITSTREAM="--offline ${BITSTREAM_KEY}"
```

Reply:
Yeah I talked with @a-will earlier today. His approach above would also work, and as @a-will pointed out to me earlier today, its what I did a while back in the airgapped env too. I think @a-will's approach here is easier, and doesn't require and any extra gitignores / Bazel rules.

Reply:
You would just need a `${BAZEL_BITSTREAMS_CACHE}/latest.txt` in your newly-created cache. The Python script at `rules/scripts/bazel_workspace.py` currently reaches for the file regardless of whether you asked for the "latest" bitstreams.

Reply:
> You would just need a `${BAZEL_BITSTREAMS_CACHE}/latest.txt` in your newly-created cache. The Python script at `rules/scripts/bazel_workspace.py` currently reaches for the file regardless of whether you asked for the "latest" bitstreams.

Ah, I see. I think explaining that in the guide (and requiring users to do it) is about at least as much work as 'fixing' this; so here's my proposal: https://github.com/lowRISC/opentitan/pull/18739

> Oh, and it turns out I had the hierarchy wrong, sorry! https://github.com/lowRISC/opentitan/blob/master/rules/scripts/bitstreams_workspace.py#L338
> 
> The directory entry with the git hash name must be under the "cache" folder.

Thanks, with those changes it works :+1: